--- UMFPACK/Demo/Makefile.orig	2012-06-21 07:39:46.000000000 +0200
+++ UMFPACK/Demo/Makefile	2012-08-28 20:34:10.400321229 +0200
@@ -13,7 +13,7 @@
 
 all64: libs run hb fortran64
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -22,69 +22,37 @@
 CONFIG1 = -DNCHOLMOD
 CONFIG2 = -DNPARTITION
 ifeq (,$(findstring -DNCHOLMOD, $(UMFPACK_CONFIG)))
-    # CHOLMOD is requested.  See if it is available
-    ifeq (../../CHOLMOD, $(wildcard ../../CHOLMOD))
-        ifeq (../../COLAMD, $(wildcard ../../COLAMD))
-            # CHOLMOD and COLAMD are available
+    # CHOLMOD is requested.
+    ifeq ($(WITH_CHOLMOD), 1)
+            # assume COLAMD available
             CONFIG1 =
-            LIB_WITH_CHOLMOD = ../../CHOLMOD/Lib/libcholmod.a \
-                ../../COLAMD/Lib/libcolamd.a
+            LIB_WITH_CHOLMOD = -lcholmod -lcolamd
             # check if METIS is requested and available
             ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
-                # METIS is requested.  See if it is available
-                ifeq ($(METIS_PATH), $(wildcard $(METIS_PATH)))
-                    ifeq (../../CAMD, $(wildcard ../../CAMD))
-                        ifeq (../../CCOLAMD, $(wildcard ../../CCOLAMD))
-                            # METIS, CAMD, and CCOLAMD are available
-                            LIB_WITH_PARTITION = $(METIS) \
-                                ../../CCOLAMD/Lib/libccolamd.a \
-                                ../../CAMD/Lib/libcamd.a
-                            I_WITH_PARTITION = -I$(METIS_PATH)/Lib \
-                                -I../../CCOLAMD/Include -I../../CAMD/Include
+                # METIS is requested.
+                ifeq ($(WITH_METIS), 1)
+                            LIB_WITH_PARTITION = -lmetis -lccolamd -lcamd
+                            I_WITH_PARTITION = -I/usr/include/metis \
+                                -I/usr/include/ccolamd -I/usr/include/camd
                             CONFIG2 =
-                        endif
-                    endif
                 endif
             endif
-        endif
     endif
 endif
 #-------------------------------------------------------------------------------
 
 C = $(CC) $(CF) $(UMFPACK_CONFIG) $(CONFIG1) $(CONFIG2) \
-    -I../Include -I../../AMD/Include -I../../SuiteSparse_config
+    -I../Include -I/usr/include/amd
 
-INC = ../Include/umfpack.h ../../AMD/Include/amd.h ../../SuiteSparse_config/SuiteSparse_config.h
+INC = ../Include/umfpack.h
 
 LIBS = $(BLAS) $(XERBLA) $(LIB) $(LIB_WITH_CHOLMOD) $(LIB_WITH_PARTITION)
 
 ../Lib/libumfpack.a:
 	( cd ../Lib ; $(MAKE) )
 
-../../AMD/Lib/libamd.a:
-	( cd ../../AMD ; $(MAKE) library )
-
-../../SuiteSparse_config/libsuitesparseconfig.a:
-	( cd ../../SuiteSparse_config ; $(MAKE) library )
-
-../../CHOLMOD/Lib/libcholmod.a:
-	- ( cd ../../CHOLMOD && $(MAKE) library )
-
-../../COLAMD/Lib/libcolamd.a:
-	- ( cd ../../COLAMD && $(MAKE) library )
-
-
-../../CCOLAMD/Lib/libccolamd.a:
-	- ( cd ../../CCOLAMD && $(MAKE) library )
-
-../../CAMD/Lib/libcamd.a:
-	- ( cd ../../CAMD && $(MAKE) library )
-
-$(METIS):
-	( cd $(METIS_PATH) && $(MAKE) )
-
-UMFPACK = ../Lib/libumfpack.a ../../AMD/Lib/libamd.a \
-    ../../SuiteSparse_config/libsuitesparseconfig.a \
+UMFPACK = ../Lib/libumfpack.a -lamd \
+    -lsuitesparseconfig \
     $(LIB_WITH_CHOLMOD) $(LIB_WITH_PARTITION)
 
 libs: $(UMFPACK)
diff -ur UMFPACK/Doc/Makefile UMFPACK-uf/Doc/Makefile
--- UMFPACK/Doc/Makefile	2006-01-30 17:51:24.000000000 +0100
+++ UMFPACK-uf/Doc/Makefile	2007-12-02 19:19:00.000000000 +0100
@@ -7,7 +7,7 @@
 
 default: dist
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
--- UMFPACK/Lib/GNUmakefile.orig	2012-05-28 01:23:50.000000000 +0200
+++ UMFPACK/Lib/GNUmakefile	2012-06-27 20:04:57.011651575 +0200
@@ -4,14 +4,14 @@
 
 default: libumfpack.a
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 # UMFPACK can use CHOLMOD by default as an ordering option
 ifneq (,$(findstring -DNCHOLMOD, $(UMFPACK_CONFIG)))
-    I = -I../Include -I../Source -I../../AMD/Include -I../../SuiteSparse_config
+    I = -I../Include -I../Source -I/usr/include/amd
 else
-    I = -I../Include -I../Source -I../../AMD/Include \
-        -I../../SuiteSparse_config -I../../CHOLMOD/Include
+    I = -I../Include -I../Source -I/usr/include/amd \
+        -I/usr/include/cholmod
 endif
 
 C = $(CC) $(CF) $(UMFPACK_CONFIG) $(I)
@@ -75,7 +75,7 @@
 # include files:
 #-------------------------------------------------------------------------------
 
-INC = ../Include/umfpack.h ../../SuiteSparse_config/SuiteSparse_config.h \
+INC = ../Include/umfpack.h \
 	../Source/umf_config.h ../Source/umf_version.h \
 	../Source/umf_internal.h ../Source/umf_triplet.h \
 	$(addprefix ../Source/,  $(addsuffix .h,$(UMFCH))) \
diff -ur UMFPACK/Lib/Makefile UMFPACK-uf/Lib/Makefile
--- UMFPACK/Lib/Makefile	2007-05-05 21:50:23.000000000 +0200
+++ UMFPACK-uf/Lib/Makefile	2007-12-02 19:19:20.000000000 +0100
@@ -8,7 +8,7 @@
 
 default: everything
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 C = $(CC) $(CFLAGS) $(UMFPACK_CONFIG) -I../Include -I../../AMD/Include \
     -I../Source
diff -ur UMFPACK/Makefile UMFPACK-uf/Makefile
--- UMFPACK/Makefile	2007-10-30 23:03:55.000000000 +0100
+++ UMFPACK-uf/Makefile	2007-12-02 19:21:14.000000000 +0100
@@ -6,7 +6,7 @@
 
 default: library
 
-include ../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 # compile all C code (except hb, fortran, and fortran64), including AMD and the
 # MATLAB mexFunctions
--- UMFPACK/MATLAB/Makefile.orig	2009-11-11 21:10:02.000000000 +0100
+++ UMFPACK/MATLAB/Makefile	2010-10-13 09:39:55.187740205 +0200
@@ -5,7 +5,7 @@
 default:
 	- echo "Use umfpack_make.m in MATLAB to compile the UMFPACK mexFunction"
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
diff -ur UMFPACK/Tcov/Makefile UMFPACK-uf/Tcov/Makefile
--- UMFPACK/Tcov/Makefile	2006-12-01 13:47:48.000000000 +0100
+++ UMFPACK-uf/Tcov/Makefile	2007-12-02 19:21:03.000000000 +0100
@@ -1,5 +1,5 @@
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 linux: distclean
 	./DO.linux
