--- UMFPACK/Demo/Makefile.orig	2011-12-14 18:45:15.000000000 +0100
+++ UMFPACK/Demo/Makefile	2012-01-06 08:03:34.253327243 +0100
@@ -13,7 +13,7 @@
 
 all64: libs run hb fortran64
 
-include ../../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -23,35 +23,28 @@
 LIB_WITH_CHOLMOD =
 ifeq (,$(findstring -DNCHOLMOD, $(UMFPACK_CONFIG)))
     # CHOLMOD is requested.  See if it is available
-    ifeq (../../CHOLMOD, $(wildcard ../../CHOLMOD))
-        ifeq (../../COLAMD, $(wildcard ../../COLAMD))
+    ifeq ($(WITH_CHOLMOD), 1)
+        # assume COLAMD available
             # CHOLMOD and COLAMD are available
             CONFIG1 =
-            LIB_WITH_CHOLMOD = ../../CHOLMOD/Lib/libcholmod.a \
-                ../../COLAMD/Lib/libcolamd.a
+            LIB_WITH_CHOLMOD = -lcholmod -lcolamd
             # check if METIS is requested and available
             ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
                 # METIS is requested.  See if it is available
-                ifeq ($(METIS), $(wildcard $(METIS)))
-                    ifeq (../../CAMD, $(wildcard ../../CAMD))
-                        ifeq (../../CCOLAMD, $(wildcard ../../CCOLAMD))
+                ifeq ($(WITH_METIS), 1)
+                    # assume CAMD and CCOLAMD are available
                             CONFIG2 =
-                            LIB_WITH_PARTITION = $(METIS) \
-                                ../../CCOLAMD/Lib/libccolamd.a \
-                                ../../CAMD/Lib/libcamd.a
-                        endif
-                    endif
+                            LIB_WITH_PARTITION = $(METIS) -lccolamd -lcamd
                 endif
             endif
-        endif
     endif
 endif
 #-------------------------------------------------------------------------------
 
 C = $(CC) $(CF) $(UMFPACK_CONFIG) $(CONFIG1) $(CONFIG2) \
-    -I../Include -I../../AMD/Include -I../../UFconfig
+    -I../Include -I/usr/include/amd
 
-INC = ../Include/umfpack.h ../../AMD/Include/amd.h ../../UFconfig/UFconfig.h
+INC = ../Include/umfpack.h
 
 LIBS = $(BLAS) $(XERBLA) $(LIB) $(LIB_WITH_CHOLMOD) $(LIB_WITH_PARTITION)
 
diff -ur UMFPACK/Doc/Makefile UMFPACK-uf/Doc/Makefile
--- UMFPACK/Doc/Makefile	2006-01-30 17:51:24.000000000 +0100
+++ UMFPACK-uf/Doc/Makefile	2007-12-02 19:19:00.000000000 +0100
@@ -7,7 +7,7 @@
 
 default: dist
 
-include ../../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
--- UMFPACK/Lib/GNUmakefile.orig	2011-12-08 19:48:30.000000000 +0100
+++ UMFPACK/Lib/GNUmakefile	2012-01-06 08:36:15.313392908 +0100
@@ -4,14 +4,14 @@
 
 default: libumfpack.a
 
-include ../../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 # UMFPACK can use CHOLMOD by default as an ordering option
 ifneq (,$(findstring -DNCHOLMOD, $(UMFPACK_CONFIG)))
-    I = -I../Include -I../Source -I../../AMD/Include -I../../UFconfig
+    I = -I../Include -I../Source -I/usr/include/amd
 else
-    I = -I../Include -I../Source -I../../AMD/Include -I../../UFconfig \
-        -I../../CHOLMOD/Include
+    I = -I../Include -I../Source -I/usr/include/amd \
+        -I/usr/include/cholmod
 endif
 
 C = $(CC) $(CF) $(UMFPACK_CONFIG) $(I)
@@ -72,7 +72,7 @@
 # include files:
 #-------------------------------------------------------------------------------
 
-INC = ../Include/umfpack.h ../../UFconfig/UFconfig.h \
+INC = ../Include/umfpack.h \
 	../Source/umf_config.h ../Source/umf_version.h \
 	../Source/umf_internal.h ../Source/umf_triplet.h \
 	$(addprefix ../Source/,  $(addsuffix .h,$(UMFCH))) \
diff -ur UMFPACK/Lib/Makefile UMFPACK-uf/Lib/Makefile
--- UMFPACK/Lib/Makefile	2007-05-05 21:50:23.000000000 +0200
+++ UMFPACK-uf/Lib/Makefile	2007-12-02 19:19:20.000000000 +0100
@@ -8,7 +8,7 @@
 
 default: everything
 
-include ../../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 C = $(CC) $(CFLAGS) $(UMFPACK_CONFIG) -I../Include -I../../AMD/Include \
     -I../Source
diff -ur UMFPACK/Makefile UMFPACK-uf/Makefile
--- UMFPACK/Makefile	2007-10-30 23:03:55.000000000 +0100
+++ UMFPACK-uf/Makefile	2007-12-02 19:21:14.000000000 +0100
@@ -6,7 +6,7 @@
 
 default: library
 
-include ../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 # compile all C code (except hb, fortran, and fortran64), including AMD and the
 # MATLAB mexFunctions
--- UMFPACK/MATLAB/Makefile.orig	2009-11-11 21:10:02.000000000 +0100
+++ UMFPACK/MATLAB/Makefile	2010-10-13 09:39:55.187740205 +0200
@@ -5,7 +5,7 @@
 default:
 	- echo "Use umfpack_make.m in MATLAB to compile the UMFPACK mexFunction"
 
-include ../../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
diff -ur UMFPACK/Tcov/Makefile UMFPACK-uf/Tcov/Makefile
--- UMFPACK/Tcov/Makefile	2006-12-01 13:47:48.000000000 +0100
+++ UMFPACK-uf/Tcov/Makefile	2007-12-02 19:21:03.000000000 +0100
@@ -1,5 +1,5 @@
 
-include ../../UFconfig/UFconfig.mk
+include /usr/share/misc/UFconfig.mk
 
 linux: distclean
 	./DO.linux
